gazebo_ros/src/gazebo_ros_api_plugin.cpp

```diff
@@ -2660,6 +2660,8 @@ bool GazeboRosApiPlugin::spawnAndConform(TiXmlDocument &gazebo_model_xml, const
 
   /// \brief poll and wait, verify that the model is spawned within Hardcoded 10 seconds
   ros::Duration model_spawn_timeout(10.0);
+  /// Wait until time can be read.
+  while (ros::Time::now().is_zero());
```

Comment:
If simulation is started paused, this line will hang until simulation starts running. I'm not sure if this is desired.

Reply:
It might make more sense to switch this to be wall time not ROS time. I'm assuming this is resource contention based and not simulation time passing. Then we won't need to worry about whether ROS time is initialized.

Reply:
As far as I understand the comment form @spark4160 it is related to simulation time passing.

Reply:
Where would I find the file gazebo_ros_api_plugin?

Reply:
https://github.com/ros-simulation/gazebo_ros_pkgs/blob/kinetic-devel/gazebo_ros/src/gazebo_ros_api_plugin.cpp

Reply:
I installed gazebo ros packages using <sudo apt-get install ros-melodic-gazebo-ros-pkgs ros-melodic-gazebo-ros-control> and I can not find this directory!
![image](https://user-images.githubusercontent.com/45830803/121805727-b0292780-cc4c-11eb-94e1-ada8f086d612.png)

Where would I find the file on melodic distribution?

Reply:
@hichamhendy did you find a solution for this

Reply:
No, I skipped it!!

Reply:
I checked the terminal that launched my gazebo world and found the errors that I needed to fix there. So if the URDF file has any issues, it doesn't load and times out.